src/docfx/config/DocumentIdConfig.cs

```diff
@@ -14,5 +21,71 @@ internal class DocumentIdConfig
         /// The site base path, used by docs.com, for backward compatibility
         /// </summary>
         public string SiteBasePath = string.Empty;
+
+        /// <summary>
+        /// The mappings between depot and files/directory
+        /// Used for backward compatibility
+        /// </summary>
+        public Dictionary<string, List<string>> DepotMappings = new Dictionary<string, List<string>>();
+
+        /// <summary>
+        /// The mappings between directory and files/directory
+        /// Used for backward compatibility
+        /// </summary>
+        public Dictionary<string, List<string>> DirectoryMappings = new Dictionary<string, List<string>>();
+
+        public DocumentIdConfig()
+        {
+            _reversedDepotMappings = new Lazy<Dictionary<string, string>>(() => ReverseAndNormalizeMappings(DepotMappings));
+            _reversedDirectoryMappings = new Lazy<Dictionary<string, string>>(() => ReverseAndNormalizeMappings(DirectoryMappings));
+        }
+
+        public (string depotName, string pathRelativeToSourceBasePath) GetMapping(string normalizedFilePathToSourceBasePath)
+        {
+            var (depotName, _) = GetReversedMapping(_reversedDepotMappings.Value, normalizedFilePathToSourceBasePath);
+            var (mappedDir, matchedDir) = GetReversedMapping(_reversedDirectoryMappings.Value, normalizedFilePathToSourceBasePath);
+
+            var mappedPathRelativeToSourceBasePath = !string.IsNullOrEmpty(matchedDir)
+                ? PathUtility.NormalizeFile(Path.Combine(mappedDir, Path.GetRelativePath(matchedDir, normalizedFilePathToSourceBasePath)))
+                : normalizedFilePathToSourceBasePath;
+
+            return (depotName, mappedPathRelativeToSourceBasePath);
+        }
+
+        private static (string value, string matchedDirectory) GetReversedMapping(Dictionary<string, string> mappings, string normalizedFilePathToSourceBasePath)
+        {
+            foreach (var (path, value) in mappings)
+            {
+                if (string.Equals(path, normalizedFilePathToSourceBasePath, PathUtility.PathComparison))
```

Comment:
we probably need a `PathUtility.PathMatches` method to detect if a path is inside a folder or is the same path. `routes` also need this.

Reply:
Also, it is proved problematic to indicate folders using trailing slash, https://github.com/dotnet/docfx/issues/3077.

Reply:
OK, i will add one.

---
In reply to: [203582744](https://github.com/dotnet/docfx/pull/3095#discussion_r203582744) [](ancestors = 203582744)

Reply:
Let's resolve this issue together, for this PR, i follow the original rules, folder need end with '/'

---
In reply to: [203583022](https://github.com/dotnet/docfx/pull/3095#discussion_r203583022) [](ancestors = 203583022)